openmc/mgxs/library.py

```diff
@@ -717,3 +721,292 @@ def load_from_file(filename='mgxs', directory='mgxs'):
 
         # Load and return pickled Library object
         return pickle.load(open(full_filename, 'rb'))
+
+    def get_xsdata(self, domain, domain_name, nuclide='total', xs_type='macro',
+                   xs_id='1m', order=-1):
+        """Generates an openmc.XSdata object describing a multi-group cross section
+        data set for eventual combination in to an openmc.MGXSLibrary object
+        (i.e., the library).
+
+        Parameters
+        ----------
+        domain : openmc.Material or openmc.Cell or openmc.Universe
+            The domain for spatial homogenization
+        domain_name : str
```

Comment:
This is a little confusing to me, as at first I thought it refers to the name of the above domain rather than the name applied to the created material. I think simply "name" would be more fitting.